### PR TITLE
r/aws_eks_cluster: Add support for EKS Hybrid nodes

### PR DESCRIPTION
**Pull request by bryantbiggs**
### Description
- Add support for EKS Hybrid nodes

:warning: Requires #40380

### Relations

Closes https://github.com/hashicorp/terraform-provider-aws/issues/40372.

### References

- https://aws.amazon.com/blogs/aws/use-your-on-premises-infrastructure-in-amazon-eks-clusters-with-amazon-eks-hybrid-nodes/
- https://aws.amazon.com/eks/hybrid-nodes/?trk=f4bc0305-3e3d-470e-bbb5-02917c6bc4c6&sc_channel=el

### Output from Acceptance Testing

```console
make testacc PKG=eks TESTS=TestAccEKSCluster_RemoteNetwork_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/eks/... -v -count 1 -parallel 20 -run='TestAccEKSCluster_RemoteNetwork_'  -timeout 360m
2024/11/29 16:49:46 Initializing Terraform AWS Provider...
=== RUN   TestAccEKSCluster_RemoteNetwork_Node
=== PAUSE TestAccEKSCluster_RemoteNetwork_Node
=== RUN   TestAccEKSCluster_RemoteNetwork_Pod
=== PAUSE TestAccEKSCluster_RemoteNetwork_Pod
=== CONT  TestAccEKSCluster_RemoteNetwork_Node
=== CONT  TestAccEKSCluster_RemoteNetwork_Pod
--- PASS: TestAccEKSCluster_RemoteNetwork_Node (673.12s)
--- PASS: TestAccEKSCluster_RemoteNetwork_Pod (916.39s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/eks	923.312s
```

```console
make testacc PKG=eks TESTS=TestAccEKSClusterDataSource_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/eks/... -v -count 1 -parallel 20 -run='TestAccEKSClusterDataSource_'  -timeout 360m
2024/11/29 17:28:38 Initializing Terraform AWS Provider...
=== RUN   TestAccEKSClusterDataSource_basic
=== PAUSE TestAccEKSClusterDataSource_basic
=== RUN   TestAccEKSClusterDataSource_outpost
=== PAUSE TestAccEKSClusterDataSource_outpost
=== RUN   TestAccEKSClusterDataSource_remoteNetwork
=== PAUSE TestAccEKSClusterDataSource_remoteNetwork
=== CONT  TestAccEKSClusterDataSource_basic
=== CONT  TestAccEKSClusterDataSource_remoteNetwork
=== CONT  TestAccEKSClusterDataSource_outpost
    cluster_data_source_test.go:79: skipping since no Outposts found
--- SKIP: TestAccEKSClusterDataSource_outpost (0.92s)
--- PASS: TestAccEKSClusterDataSource_basic (566.37s)
--- PASS: TestAccEKSClusterDataSource_remoteNetwork (582.80s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/eks
```

### CODE REVIEW

File: website/docs/r/eks_cluster.html.markdown [hunk 1]
```diff
@@ -10,200 +10,201 @@ description: |-
 
 Manages an EKS Cluster.
 
-> **Hands-on:** For an example of `aws_eks_cluster` in use, follow the [Provision an EKS Cluster](https://learn.hashicorp.com/tutorials/terraform/eks) tutorial on HashiCorp Learn.
+-> For an example of `aws_eks_cluster` in use, follow the [Provision an EKS Cluster](https://learn.hashicorp.com/tutorials/terraform/eks) tutorial on HashiCorp Learn.
```
Comment (gdavison) [inline website/docs/r/eks_cluster.html.markdown:13]: This should stay as `> **Hands-on:**`

```suggestion
> **Hands-on:** For an example of `aws_eks_cluster` in use, follow the [Provision an EKS Cluster](https://learn.hashicorp.com/tutorials/terraform/eks) tutorial on HashiCorp Learn.
```

**Review by gdavison — CHANGES_REQUESTED** [review]
I missed a comment when reviewing

**Comment by bryantbiggs** [comment]
```console
make testacc PKG=eks TESTS=TestAccEKSCluster_RemoteNetwork_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/eks/... -v -count 1 -parallel 20 -run='TestAccEKSCluster_RemoteNetwork_'  -timeout 360m
2024/12/02 12:06:54 Initializing Terraform AWS Provider...
=== RUN   TestAccEKSCluster_RemoteNetwork_Node
=== PAUSE TestAccEKSCluster_RemoteNetwork_Node
=== RUN   TestAccEKSCluster_RemoteNetwork_Pod
=== PAUSE TestAccEKSCluster_RemoteNetwork_Pod
=== CONT  TestAccEKSCluster_RemoteNetwork_Node
=== CONT  TestAccEKSCluster_RemoteNetwork_Pod
--- PASS: TestAccEKSCluster_RemoteNetwork_Node (496.33s)
--- PASS: TestAccEKSCluster_RemoteNetwork_Pod (678.89s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/eks	686.739s
```

```console
make testacc PKG=eks TESTS=TestAccEKSClusterDataSource_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/eks/... -v -count 1 -parallel 20 -run='TestAccEKSClusterDataSource_'  -timeout 360m
2024/12/02 12:27:53 Initializing Terraform AWS Provider...
=== RUN   TestAccEKSClusterDataSource_basic
=== PAUSE TestAccEKSClusterDataSource_basic
=== RUN   TestAccEKSClusterDataSource_outpost
=== PAUSE TestAccEKSClusterDataSource_outpost
=== RUN   TestAccEKSClusterDataSource_remoteNetwork
=== PAUSE TestAccEKSClusterDataSource_remoteNetwork
=== CONT  TestAccEKSClusterDataSource_basic
=== CONT  TestAccEKSClusterDataSource_remoteNetwork
=== CONT  TestAccEKSClusterDataSource_outpost
    cluster_data_source_test.go:79: skipping since no Outposts found
--- SKIP: TestAccEKSClusterDataSource_outpost (0.90s)
--- PASS: TestAccEKSClusterDataSource_remoteNetwork (630.20s)
--- PASS: TestAccEKSClusterDataSource_basic (655.89s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/eks	661.333s
```

**Review by gdavison — APPROVED** [review]
Looks good! 🚀 

--- PASS: TestAccEKSCluster_BootstrapSelfManagedAddons_migrate (563.36s)
--- PASS: TestAccEKSCluster_zonalShiftConfig (580.70s)
--- PASS: TestAccEKSCluster_tags (622.03s)
--- PASS: TestAccEKSClusterDataSource_basic (635.57s)
--- PASS: TestAccEKSCluster_RemoteNetwork_Node (637.44s)
--- PASS: TestAccEKSCluster_RemoteNetwork_Pod (651.74s)
--- PASS: TestAccEKSCluster_Encryption_create (656.18s)
--- PASS: TestAccEKSCluster_upgradePolicy (658.09s)
--- PASS: TestAccEKSClusterDataSource_remoteNetwork (667.40s)
--- PASS: TestAccEKSCluster_basic (701.18s)
--- PASS: TestAccEKSCluster_AccessConfig_update (703.71s)
--- PASS: TestAccEKSCluster_disappears (710.53s)
--- PASS: TestAccEKSCluster_logging (732.94s)
--- PASS: TestAccEKSCluster_AccessConfig_create (733.60s)
--- PASS: TestAccEKSCluster_VPC_securityGroupIDs (496.29s)
--- PASS: TestAccEKSCluster_VPC_endpointPublicAccess (1172.23s)
--- PASS: TestAccEKSCluster_Encryption_versionUpdate (1194.02s)
--- PASS: TestAccEKSCluster_Network_ipFamily (1204.70s)
--- PASS: TestAccEKSCluster_BootstrapSelfManagedAddons_update (1253.80s)
--- PASS: TestAccEKSCluster_version (1297.37s)
--- PASS: TestAccEKSCluster_VPC_publicAccessCIDRs (801.51s)
--- PASS: TestAccEKSCluster_Encryption_update (1557.97s)
--- PASS: TestAccEKSCluster_Network_serviceIPv4CIDR (1199.98s)
--- PASS: TestAccEKSCluster_VPC_endpointPrivateAccess (1400.00s)
--- PASS: TestAccEKSCluster_VPC_securityGroupIDsAndSubnetIDs_update (1428.91s)

**Pull request merged**